roles/zuul/templates/etc/zuul/config/layout.yaml

```diff
@@ -92,8 +92,10 @@ projects:
   - name: BonnyCI/hoist
     check_github:
       - bonnyci-run-check-multinode
+      - dib-build-check
     gate_github:
       - bonnyci-run-gate-multinode
+      - dib-build-gate
 
   - name: BonnyCI/sandbox
     check_github:
```

Comment:
So we should be able to only run the dib-build-pipeline job when files in the roles/nodepool/files/etc/nodepool/elements are changed by the patch under test.  See https://docs.openstack.org/infra/zuul/zuul.html#jobs

I *think* we just need to add an entry to the layouts "jobs" section that applies the "files:" option to the dib job.